### PR TITLE
CP1/CP2 engagement rules.

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/health/careplan.rb [hunk 1]
```diff
@@ -269,8 +296,12 @@ def active?
       completed? && expires_on >= Date.current
     end
 
+    CP2_DATE = '2023-04-01'.to_date.freeze
     def completed?
-      provider_signed_on && patient_signed_on
+      cp1 = provider_signed_on && patient_signed_on && [provider_signed_on, patient_signed_on].max < CP2_DATE
+      cp2 = patient_signed_on >= CP2_DATE && rn_approval?
```
Comment (eanders) [inline app/models/health/careplan.rb]: Where is `rn_approval?` coming from.  I can't find it.  Also, does it guarantee `patient_signed_on` is present?

File: app/models/health/patient.rb [hunk 2]
```diff
@@ -279,6 +275,59 @@ class Patient < Base
       )
     end
 
+    def self.cp_2_engagement(on) # rubocop:disable Naming/MethodParameterName
```
Comment (eanders) [inline app/models/health/patient.rb:282]: Might be worth putting in a comment to explain what constitutes engaged for CP 2

File: app/models/health/patient.rb [hunk 3]
```diff
@@ -279,6 +275,59 @@ class Patient < Base
...
+      ssm_patient_id_scope = Health::SelfSufficiencyMatrixForm.distinct.
+        completed.
+        allowed_for_engagement.
+        where(completed_at: (..on.to_time)).
```
Comment (eanders) [inline app/models/health/patient.rb:286]: Should all of these date fields be limited to time after 4/1/2023?

**Review by eanders — CHANGES_REQUESTED** [review]
Couple of inline items and the tests aren't passing.  Otherwise looks good.  Thanks!

File: app/models/health/patient.rb (hunk 3 above)
Comment (awisema) [reply app/models/health/patient.rb:286]: I don't think so -- they are still valid if if they were completed before 4/1.

**Comment by awisema** [comment]
I'm deferring the test suite changes to the integration branch as there are too many moving pieces right now...

**Review by eanders — APPROVED** [review]

**Pull request merged**